frontend_tests/zjsunit/namespace.js

```diff
@@ -143,20 +137,14 @@ exports.mock_template = (fn, exercise_template) => {
 
     obj.f.__default = true;
 
-    const filename = get_validated_filename("../../static/templates/" + fn);
-
-    // We update module_mocks with our object, but load() will return
-    // its own function that calls our obj.f.
-    module_mocks.set(filename, obj);
-
-    return obj;
+    return exports.mock_cjs("../../static/templates/" + fn, obj);
 };
 
-exports.mock_esm = (fn, obj = {}) => {
+exports.mock_esm = (request, obj = {}) => {
```

Comment:
(A word of explanation, since you seem to really want to change this to `fn`: this is a module name, not a filename or a function. `request` is the name used by upstream in [`Module._resolveFilename`](https://github.com/nodejs/node/blob/v16.3.0/lib/internal/modules/cjs/loader.js#L856). I’m open to considering other names, but `fn` is wrong.)

Reply:
How about we do `module_path`?